rosette/solver/smt/yices-smt2.rkt

```diff
@@ -0,0 +1,69 @@
+#lang racket
+
+(require racket/runtime-path 
+         "server.rkt" "env.rkt" 
+         "../solver.rkt"
+         (prefix-in base/ "base-solver.rkt"))
+
+(provide (rename-out [make-yices-smt2 yices-smt2]) yices-smt2? yices-smt2-available?)
```

Comment:
Why is this called `yices-smt2` rather than e.g. `yices2` (or `yices`)?

Reply:
See here: https://github.com/SRI-CSL/yices2/blob/e27cf308cffb0ecc6cc7165c10e81ca65bc303b3/doc/NOTES#L1182

Apparently they make different binaries for each frontend (rather than changing the frontend with a flag, as other solvers do)

Reply:
I think this should be called `yices`, for consistency with other Rosette solvers.